lib/matplotlib/axis.py

```diff
@@ -1831,21 +1831,38 @@ def set_ticks(self, ticks, minor=False):
         choice to prevent the surprise of a non-visible tick. If you need
         other limits, you should set the limits explicitly after setting the
         ticks.
+
+        The ticks of any shared axis will also be updated.
         """
         # XXX if the user changes units, the information will be lost here
         ticks = self.convert_units(ticks)
-        if len(ticks) > 1:
-            xleft, xright = self.get_view_interval()
-            if xright > xleft:
-                self.set_view_interval(min(ticks), max(ticks))
+        if self is self.axes.xaxis:
+            shared = [
+                ax.xaxis
+                for ax in self.axes.get_shared_x_axes().get_siblings(self.axes)
+            ]
+        elif self is self.axes.yaxis:
+            shared = [
+                ax.yaxis
+                for ax in self.axes.get_shared_y_axes().get_siblings(self.axes)
+            ]
+        else:
+            shared = [self]
+        for axis in shared:
+            if len(ticks) > 1:
```

Comment:
It seems that maybe `xleft` and `xright` should be set outside the loop in case `xleft` and `xright` have been swapped on a shared axes somehow?  I assume the idea is for all the axes to be identical.